From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-page_isolation-remove-migratetype-parameter-from-more-functions-fix-2
Date: Fri May  9 01:48:29 PM PDT 2025

fix build warnings

> >> mm/page_isolation.c:18:20: warning: function 'get_pageblock_isolate' is not needed and will not be emitted [-Wunneeded-internal-declaration]
>       18 | static inline bool get_pageblock_isolate(struct page *page)
>          |                    ^~~~~~~~~~~~~~~~~~~~~
>    mm/page_isolation.c:28:20: warning: unused function 'set_pageblock_isolate' [-Wunused-function]
>       28 | static inline void set_pageblock_isolate(struct page *page)
>          |                    ^~~~~~~~~~~~~~~~~~~~~
>    2 warnings generated.


Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202505092126.yRGhLhg1-lkp@intel.com/
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_isolation.c |    9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/mm/page_isolation.c~mm-page_isolation-remove-migratetype-parameter-from-more-functions-fix-2
+++ a/mm/page_isolation.c
@@ -15,22 +15,19 @@
 #define CREATE_TRACE_POINTS
 #include <trace/events/page_isolation.h>
 
+#ifdef CONFIG_DEBUG_VM
 static inline bool get_pageblock_isolate(struct page *page)
 {
 	return get_pfnblock_flags_mask(page, page_to_pfn(page),
 			PB_migrate_isolate_bit);
 }
+#endif
+
 static inline void clear_pageblock_isolate(struct page *page)
 {
 	set_pfnblock_flags_mask(page, 0, page_to_pfn(page),
 			PB_migrate_isolate_bit);
 }
-static inline void set_pageblock_isolate(struct page *page)
-{
-	set_pfnblock_flags_mask(page, PB_migrate_isolate_bit,
-			page_to_pfn(page),
-			PB_migrate_isolate_bit);
-}
 
 /*
  * This function checks whether the range [start_pfn, end_pfn) includes
_
